googler

```diff
@@ -836,11 +836,12 @@ def new_connection(domain=None):
     return http.client.HTTPSConnection(domain if domain else server, timeout=45)
 
 
-def quit_program(conn):
-    """Closes a connection and quits the program"""
+def quit_program(exitcode=1):
+    """Close the connection and quits the program."""
 
+    global conn
```

Comment:
I think we should only have one open connection at any time. The old connection should be closed before the new one is opened. The old `quit_program(conn)` also closes one connection only.